In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
filepath = '../Dataset/'
train = pd.read_csv(filepath + 'Train.csv')
test = pd.read_csv(filepath + 'Test.csv')
dataDic = pd.read_csv(filepath + 'DataDictionary.csv')

In [3]:
train.shape, test.shape, dataDic.shape

((44907, 45), (1528, 6), (44, 2))

The train data has 44907 entries which is to be used to make a prediction on 1528 entries.

Interestingly, the test data has just 6 rows of data compared to the 45 rows of data in the train data.

A deeper look is needed especially with the fact that there are 44 described features in the Data Dictionary

In [4]:
test.head()

sku_name  month  year  CAT_GENDER_BOTH  CAT_GENDER_MEN  \
0  YOSHTLYNYOSHZZ     11  2021                0               1   
1  YOSHTLYNYOSHZZ     12  2021                0               1   
2  YOSHTLYNYOSHZZ      1  2022                0               1   
3  YOSHTLYNYOSHZZ      2  2022                0               1   
4    YOSHRENECARL     11  2021                0               0   

   CAT_GENDER_WOMEN  
0                 0  
1                 0  
2                 0  
3                 0  
4                 1

In [5]:
test.columns

Index(['sku_name', 'month', 'year', 'CAT_GENDER_BOTH', 'CAT_GENDER_MEN',
       'CAT_GENDER_WOMEN'],
      dtype='object')

In [6]:
train.head()

sku_name  starting_inventory  sellin  sellin_channel_1  \
0  YOSHWARDTERR                   0    1013                 0   
1  YOSHWARDTERR                   0    2026                 0   
2  YOSHWARDTERR                   0    1013                 0   
3  YOSHUANEMARX                   0  320108              4052   
4  YOSHUANEMARX                   0  132703              2026   

   sellin_channel_2  sellin_channel_3  sellin_channel_4  sellin_channel_5  \
0                 0                 0              1013                 0   
1                 0                 0              2026                 0   
2                 0                 0              1013                 0   
3             40520            240081              4052              6078   
4                 0             81040              3039             25325   

   sellin_channel_6  sellin_channel_7  ...  month  year  \
0                 0                 0  ...      7  2016   
1                 0                 0  ...      2  2017   
2                 0                 0  ...      5  2017   
3             18234                 0  ...      4  2018   
4             18234                 0  ...      5  2018   

   product_lifecycle_stage   FLAG100  disc_month  cum_disc  CAT_GENDER_BOTH  \
0                        U  0.000000           0         0                0   
1                        U  0.000000           0         0                0   
2                        U  0.000000           0         0                0   
3                        W  0.270966           1         1                0   
4                        W  0.063004           0         1                0   

   CAT_GENDER_MEN  CAT_GENDER_WOMEN  Weeks  
0               1                 0      1  
1               1                 0      0  
2               1                 0      0  
3               0                 1      1  
4               0                 1      0  

[5 rows x 45 columns]

In [7]:
train.columns

Index(['sku_name', 'starting_inventory', 'sellin', 'sellin_channel_1',
       'sellin_channel_2', 'sellin_channel_3', 'sellin_channel_4',
       'sellin_channel_5', 'sellin_channel_6', 'sellin_channel_7',
       'sellin_channel_8', 'sellout', 'onhand_inventory', 'leftover_inventory',
       'sellout_channel_1', 'sellout_channel_2', 'sellout_channel_3',
       'sellout_channel_4', 'sellout_channel_5', 'sellout_channel_6',
       'sellout_channel_7', 'sellout_channel_8', 'sellout_channel_9',
       'sellout_channel_10', 'onhand_inventory_channel_1',
       'onhand_inventory_channel_2', 'onhand_inventory_channel_3',
       'onhand_inventory_channel_4', 'onhand_inventory_channel_5',
       'onhand_inventory_channel_6', 'onhand_inventory_channel_7',
       'onhand_inventory_channel_8', 'onhand_inventory_channel_9',
       'onhand_inventory_channel_10', 'price', 'month', 'year',
       'product_lifecycle_stage', 'FLAG100', 'disc_month', 'cum_disc',
       'CAT_GENDER_BOTH', 'CAT_GENDER_MEN',

In [8]:
dataDic.head()

Column Name                                        Description
0            sku_name                                        Product SKU
1  starting_inventory  Starting inventory at the beginning of the mon...
2              sellin  The sellin or demand. What is sold in to our c...
3    sellin_channel_1                           The sellin for channel 1
4    sellin_channel_2                           The sellin for channel 2

In [9]:
dataDic.tail()

Column Name                                        Description
39          cum_disc  The effect of a discount may not be captured b...
40   CAT_GENDER_BOTH                             SKU is male and female
41    CAT_GENDER_MEN                                        SKU is male
42  CAT_GENDER_WOMEN                                      SKU is female
43             Weeks      The number of weekends that certain month had

"""
The objective is to accurately predict the “sellin”, which is the optimal inventory needed to meet demand by Fossil to its own customers or channels, four months into the future.
"""

According to the task description

In [10]:
train.sellin

0          1013
1          2026
2          1013
3        320108
4        132703
          ...  
44902    156002
44903     34442
44904     38494
44905     29377
44906     18234
Name: sellin, Length: 44907, dtype: int64

In [11]:
dataDic[dataDic['Column Name'] == 'sellin'].iloc[0,1]

'The sellin or demand. What is sold in to our customers. \nThis is what we are trying to predict 4 months from now (Dependent)'

We are trying to predict the above

In [12]:
test_col = list(test.columns)

train[test_col].head()

sku_name  month  year  CAT_GENDER_BOTH  CAT_GENDER_MEN  \
0  YOSHWARDTERR      7  2016                0               1   
1  YOSHWARDTERR      2  2017                0               1   
2  YOSHWARDTERR      5  2017                0               1   
3  YOSHUANEMARX      4  2018                0               0   
4  YOSHUANEMARX      5  2018                0               0   

   CAT_GENDER_WOMEN  
0                 0  
1                 0  
2                 0  
3                 1  
4                 1